### PR TITLE
test merge and comment assertions

### PR DESCRIPTION
**Pull request by Killakan789**
test merge

### CODE REVIEW

File: symfony/src/Dto/CustomerDTO.php [hunk 1]
```diff
@@ -43,7 +39,6 @@ final class CustomerDTO
     /**
      * @Assert\Type("string")
      * @Assert\Length(min = 8, max = 20, minMessage = "minimal length is 8 numbers in a string", maxMessage = "maximal length is 20 numbers in a string")
-     * @Assert\Regex(pattern="/^[0-9]*$/", message="this should contain only numbers")
      * @Assert\NotNull()
      * @Assert\NotBlank()
      */
```
Comment (Killakan789) [inline symfony/src/Dto/CustomerDTO.php]: Testing multiple comment (here you can see validation rules for certain property) #1

File: symfony/src/Controller/CustomerController.php [hunk 2]
```diff
@@ -58,7 +58,15 @@ public function get($id)
...
         $jsonContent = $this->serializer->serialize($customer, 'json');
-        return new Response($jsonContent, Response::HTTP_OK);
+        $dto = $this->serializer->deserialize($jsonContent, CustomerDTO::class, 'json');
+        //validate fields
+        $errors = $this->validator->validate($dto);
+        if (0 === count($errors)) {
+            return new Response($jsonContent, Response::HTTP_OK);
+        }else{
+            $jsonContent = $this->serializer->serialize($errors,'json');
+            return new Response($jsonContent, Response::HTTP_OK);
```
Comment (avionwd) [inline symfony/src/Controller/CustomerController.php:68]: Response with errors can't have response code `Response::HTTP_OK` (200 OK). It should be `Response:: HTTP_BAD_REQUEST` (400 Bad Request). Take a look into available response codes: https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/HttpFoundation/Response.php#L23

File: symfony/src/Dto/CustomerDTO.php [hunk 3]
```diff
@@ -43,7 +39,6 @@ final class CustomerDTO
     /**
      * @Assert\Type("string")
      * @Assert\Length(min = 8, max = 20, minMessage = "minimal length is 8 numbers in a string", maxMessage = "maximal length is 20 numbers in a string")
```
Comment (avionwd) [inline symfony/src/Dto/CustomerDTO.php]: nit: message can be omitted as the default message contains all necessary description and also translated into several languages.

File: symfony/src/Dto/CustomerDTO.php [hunk 4]
```diff
@@ -6,15 +6,11 @@
...
 final class CustomerDTO
 {
     /**
-     * @Assert\Type(
-     *     type="integer",
-     *     message="The value {{ value }} is not a valid {{ type }}."
-     * )
+     * @Assert\Type("integer")
      */
     public $id;
```
Comment (avionwd) [inline symfony/src/Dto/CustomerDTO.php:15]: If properties defined as public in the class there is no need to create getters of setter, as you can always get the value from the property itself.

File: symfony/src/Controller/CustomerController.php (hunk 2 above)
Comment (avionwd) [inline symfony/src/Controller/CustomerController.php]: Response with errors can't have response code `Response::HTTP_OK` (200 OK). It should be `Response:: HTTP_BAD_REQUEST` (400 Bad Request). Take a look into available response codes: https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/HttpFoundation/Response.php#L23

**Review by avionwd — CHANGES_REQUESTED** [review]
Well for issue #1 it's ok to merge this stuff. Fix comments and we can merge changes into the main branch.

File: symfony/src/Controller/CustomerController.php [hunk 5]
```diff
@@ -42,7 +42,7 @@ public function add(Request $request)
             return new Response($jsonContent, Response::HTTP_OK);
         }else{
             $jsonContent = $this->serializer->serialize($errors,'json');
-            return new Response($jsonContent, Response::HTTP_OK);
+            return new Response($jsonContent, Response::HTTP_BAD_REQUEST);
```
Comment (Killakan789) [inline symfony/src/Controller/CustomerController.php:45]: changed response to 400 request #1

File: symfony/src/Dto/CustomerDTO.php [hunk 6]
```diff
@@ -38,7 +38,7 @@ final class CustomerDTO
     public $email;
     /**
      * @Assert\Type("string")
-     * @Assert\Length(min = 8, max = 20, minMessage = "minimal length is 8 numbers in a string", maxMessage = "maximal length is 20 numbers in a string")
+     * @Assert\Length(min = 8, max = 20)
```
Comment (Killakan789) [inline symfony/src/Dto/CustomerDTO.php:41]: disabled message cause actually we have default one and it works as it should (just practiced with asserts)

**Pull request merged**